src/blob_file_iterator.cc

```diff
@@ -48,7 +44,26 @@ bool BlobFileIterator::Init() {
   BlobFileFooter blob_file_footer;
   status_ = blob_file_footer.DecodeFrom(&slice);
   end_of_blob_record_ = file_size_ - BlobFileFooter::kEncodedLength -
-                        blob_file_footer.meta_index_handle.size();
+                        blob_file_footer.meta_index_handle.size() -
+                        kBlockTrailerSize;
```

Comment:
It seems if meta_index_handle is empty, we should not subtract `kBlockTrailerSize`?

Reply:
yeah, will fix it